dwave/system/composites/linear_ancilla.py

```diff
@@ -0,0 +1,213 @@
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""Embedding composite to implement linear fields as polarized ancilla qubits.
+"""
+
+import numbers
+
+from typing import Sequence, Mapping, Any
+
+from collections import defaultdict
+
+import numpy as np
+
+from dimod.decorators import nonblocking_sample_method
+import dimod
+
+
+__all__ = ["LinearAncillaComposite"]
+
+
+class LinearAncillaComposite(dimod.ComposedSampler, dimod.Structured):
+    """Implements linear fields as polarized ancilla qubits.
+
+    Linear field `h_i` of qubit `i` is implemented through a coupling `J_{ij}` between
+    the qubit and a neighbouring qubit `j` that is fully polarized with a large flux bias.
+
+    Args:
+    child_sampler (:class:`dimod.Sampler`):
+        A dimod sampler, such as a :obj:`DWaveSampler`, that has flux bias controls.
+
+    """
+
+    def __init__(
+        self,
+        child_sampler: dimod.Sampler,
+    ):
+        self.children = [child_sampler]
+        self.parameters = child_sampler.parameters.copy()
+        self.properties = dict(child_properties=child_sampler.properties.copy())
+        self.nodelist = child_sampler.nodelist
+        self.edgelist = child_sampler.edgelist
+
+    def nodelist(self):
+        pass  # overwritten by init
+
+    def edgelist(self):
+        pass  # overwritten by init
+
+    children = None  # overwritten by init
+    """list [child_sampler]: List containing the structured sampler."""
+
+    parameters = None  # overwritten by init
+    """dict[str, list]: Parameters in the form of a dict.
+
+    For an instantiated composed sampler, keys are the keyword parameters
+    accepted by the child sampler and parameters added by the composite.
+    """
+
+    properties = None  # overwritten by init
+    """dict: Properties in the form of a dict.
+
+    Contains the properties of the child sampler.
+    """
+
+    @nonblocking_sample_method
+    def sample(
+        self,
+        bqm: dimod.BinaryQuadraticModel,
+        *,
+        h_tolerance: numbers.Number = 0,
+        default_flux_bias_range: Sequence[float] = [-0.005, 0.005],
+        **parameters,
+    ):
+        """Sample from the provided binary quadratic model.
+
+
+        Args:
+            bqm (:obj:`~dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            h_tolerance (:class:`numbers.Number`):
+                Magnitude of the linear bias can be left on the qubit. Assumed to be positive. Defaults to zero.
```

Comment:
```suggestion
                Maximum linear bias to be set directly on problem qubits; above this the bias
                is emulated by the flux-bias offset to an ancilla qubit. Assumed to be positive. 
                Defaults to zero.
```

Reply:
Keeping "magnitude" because I am looking for the maximum absolute. Hence assumed to be positive.
```
                Magnitude of the linear bias to be set directly on problem qubits; above this the bias
                is emulated by the flux-bias offset to an ancilla qubit. Assumed to be positive. 
                Defaults to zero.
```